Server/ConcordiaCurriculumManager/Services/CourseService.cs

```diff
@@ -111,6 +112,31 @@ public async Task<CourseModificationRequest> InitiateCourseModification(CourseMo
         return courseModificationRequest;
     }
 
+    public async Task<CourseDeletionRequest> InitiateCourseDeletion(CourseDeletionInitiationDTO deletion, Guid userId)
+    {
+        var oldCourse = await _courseRepository.GetCourseByCourseId(deletion.CourseId) ?? throw new ArgumentException("The course does not exist.");
+
+        Dossier dossier = await _dossierService.GetDossierForUserOrThrow(deletion.DossierId, userId);
+
+        var newDeletedCourse = CreateCourseFromDTOData(deletion, oldCourse.CourseID, oldCourse.Version + 1);

```

Comment:
I think it's much simpler here to just make a clone of oldCourse and then do:
```
oldCourseClone.Version++;
oldCourseClone.CourseStateEnum = CourseStateEnum.CourseDeletionProposal;
```
This ties into the proposal to simplify the DTO, since there's a lot of unnecessary data being sent. Potentially, this cloning then modification of data can be made a method in Course itself.

Reply:
Sounds good.